### PR TITLE
qtile: add missing python package dependencies

### PR DESCRIPTION
**Pull request by vividn**
###### Motivation for this change
Qtile cannot connect to dbus and cause applications to freeze when they attempt to send notifications.
For example, slack will completely freeze when a new message comes in after it attempts to send a notification.

This is caused by an inability to import dbus and pygobject3 as evidenced by this error in the logs
`WARNING libqtile manager.py:setup_eventloop():L240 importing dbus/gobject failed, dbus will not work.`

Also added the python packages dependencies mentioned here:
#45039
DavHau/mach-nix#125

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by kamilchm — APPROVED** [review]

**Comment by vividn** [comment]
Hey @kamilchm, this is my first PR on nixpkgs. Is there anything else I have to do here before this gets merged in?

**Comment by kamilchm** [comment]
Hi, anything we can do now is waiting for someone with merge permission, which should happen shortly.

**Comment by vividn** [comment]
I notice that this PR doesn't appear when searching for "review:approved" in the PR list. Probably because I marked it ready to review after you already approved it. Would you mind reapproving it?

**Review by kamilchm — APPROVED** [review]

**Comment by vividn** [comment]
Hey @Lassulus, I noticed you were the last committer for qtile. Could you review and merge this? The changes are minimal. Sorry if this is the wrong way to get it merged, but it's been a few weeks, and I'm worried that this PR will drift into backlog-hell.

**Comment by Lassulus** [comment]
@vividn sure.
I guess its an okay way to get attention. Maybe you can add me to reviewers next time, then I will get notified by github. Stuff could still fall under the rug though :D

**Comment by vividn** [comment]
Thanks!
As far as I could tell, I didn't have permissions to add reviewers. The gear icon wasn't appearing for me--it seems entirely automated by bots.

**Pull request merged**
